lib/on_message.js

```diff
@@ -0,0 +1,14 @@
+var on_message = function(haxfred, message) {
+  if (message.type === 'message') {
+    var type = isDirect(haxfred.slack.self.id, message.text) || message.subtype || message.type;
+    haxfred.emit('slack.' + type, message);
+  }
+}
+
+function isDirect(userId, messageText) {
+  var userTag = '<@' + userId + '>';
+  return messageText &&
+    messageText.indexOf(userTag) > -1 ? 'directMsg' : false;
+}
+
+module.exports = on_message;
```

Comment:
I really like how breaking this out makes things more testable, but it feels like something that could be contained within the base (just exported as a `_`ivitized method or an export itself). I don't have a super strong rationale for that, but it seems like this isn't really a separate "module" in of itself, more of a handler that lives inside of something else. Totally up for discussion.